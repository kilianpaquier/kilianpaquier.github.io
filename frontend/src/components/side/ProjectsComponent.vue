<template>
    <v-row :no-gutters="true">
        <v-col cols="12">
            <h2 class="mb-3">{{ $t("projects_title") }}</h2>
        </v-col>
        <v-col cols="12">
            <v-window :continuous="true" class="elevation-1" show-arrows="hover">
                <v-window-item v-for="(project, index) in projects" :key="'project_' + index" eager>
                    <v-card>
                        <v-img :eager="true" :src="project.icon" class="align-end cursor-pointer" @click="sectionShowing = !sectionShowing">
                            <v-card-title v-if="project.title" :class="project.titleColor">
                                {{ project.title }}
                            </v-card-title>
                        </v-img>

                        <v-expand-transition>
                            <div v-show="sectionShowing">
                                <v-card-text>
                                    <v-row :no-gutters="true">
                                        <v-col v-for="(text, index) in project.text" :key="'project_text' + index" class="text-center my-1" cols="12">
                                            {{ $t(text) }}
                                        </v-col>

                                        <v-col class="text-center my-3" cols="12">
                                            <v-img :eager="true" :src="project.codeImg" class="mx-auto cursor-pointer" height="50" @click="redirect(index)"></v-img>
                                        </v-col>

                                        <template v-if="project.captions">
                                            <v-col v-for="(caption, index) in project.captions" :key="'project_caption' + index" class="text-caption text-end" cols="12">
                                                {{ $t(caption) }}
                                            </v-col>
                                        </template>
                                    </v-row>
                                </v-card-text>
                            </div>
                        </v-expand-transition>
                    </v-card>
                </v-window-item>
            </v-window>
        </v-col>
    </v-row>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import {projects} from "@/data/projects";

export default defineComponent({
    name: "ProjectsComponent",

    data: () => ({
        sectionShowing: false
    }),

    setup: () => ({
        projects,
        redirect: (index: number) => window.open(projects[index].codeLink)
    })
});
</script>