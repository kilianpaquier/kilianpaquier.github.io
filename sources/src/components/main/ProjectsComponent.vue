<template>
  <v-container fluid>
    <!-- Projects title -->
    <h2
      class="font-weight-bold mb-3"
      v-text="$vuetify.lang.t(`$vuetify.projects-title`)"
    />

    <!-- Window component with all the projects -->
    <v-window
      v-model="window"
      class="elevation-1"
      continuous
      horizontal
      show-arrows
      show-arrows-on-hover
    >
      <!-- Previous arrow -->
      <template v-slot:prev="{ on, attrs }">
        <v-btn color="white" icon v-bind="attrs" v-on="on">
          <v-icon v-text="mdiChevronLeft" />
        </v-btn>
      </template>

      <!-- Next arrow -->
      <template v-slot:next="{ on, attrs }">
        <v-btn color="white" icon v-bind="attrs" v-on="on">
          <v-icon v-text="mdiChevronRight" />
        </v-btn>
      </template>

      <!-- Window item per project -->
      <v-window-item v-for="(project, i) in projects" :key="i" eager>
        <v-card>
          <!-- Main component, the img -->
          <v-img
            :src="require(`@/assets/projects/${project.img}`)"
            class="align-end cursor-pointer"
            eager
            @click="sectionShowing = !sectionShowing"
          >
            <!-- Project title if needed -->
            <v-card-title
              v-if="project.title"
              :class="project.titleColor"
              v-text="project.title"
            />

            <!-- Row with X buttons to change the selected project -->
            <v-row class="pb-1">
              <v-col class="d-flex" cols="12">
                <div class="mx-auto d-none d-sm-block">
                  <v-item-group v-model="window" mandatory>
                    <v-item
                      v-for="(list, i) in projects"
                      :key="i"
                      v-slot="{ active, toggle }"
                    >
                      <v-btn
                        :color="project.arrowsColor"
                        :input-value="active"
                        icon
                        @click.stop="toggle"
                      >
                        <v-icon v-text="mdiRecord" />
                      </v-btn>
                    </v-item>
                  </v-item-group>
                </div>
              </v-col>
            </v-row>
          </v-img>

          <!-- Expansion panel with a summary of the project -->
          <v-expand-transition>
            <div v-show="sectionShowing">
              <v-card-text>
                <!-- Showing multiple paragraphs -->
                <v-row>
                  <v-col
                    cols="12"
                    class="py-2"
                    v-for="(text, i) in project.text"
                    :key="i"
                    v-text="$vuetify.lang.t(`$vuetify.${text}`)"
                  />
                </v-row>

                <v-row>
                  <!-- Img or Icon for the git link -->
                  <v-col
                    cols="12"
                    class="text-center py-2"
                    :class="{ 'mb-4': project.captions }"
                  >
                    <v-img
                      v-if="project.codeImg"
                      :src="require(`@/assets/projects/${project.codeImg}`)"
                      class="mx-auto cursor-pointer"
                      width="50"
                      @click="redirectTo(project.codeLink)"
                      eager
                    />

                    <a
                      v-if="project.codeIcon"
                      @click="redirectTo(project.codeLink)"
                    >
                      <em
                        :class="project.codeIcon"
                        class="colored skill-icon"
                      />
                    </a>
                  </v-col>
                </v-row>

                <!-- Captions if needed -->
                <v-row class="text-right" no-gutters v-if="project.captions">
                  <v-col
                    cols="12"
                    class="caption"
                    v-for="(caption, i) in project.captions"
                    :key="i"
                    v-text="$vuetify.lang.t(`$vuetify.${caption}`)"
                  />
                </v-row>
              </v-card-text>
            </div>
          </v-expand-transition>
        </v-card>
      </v-window-item>
    </v-window>
  </v-container>
</template>

<script lang="ts">
import { mdiChevronLeft, mdiChevronRight, mdiRecord } from "@mdi/js";
import Vue from "vue";
import { Project } from "@/model/entities";
import Component from "vue-class-component";

@Component
export default class ProjectsComponent extends Vue {
  mdiChevronRight = mdiChevronRight;
  mdiChevronLeft = mdiChevronLeft;
  mdiRecord = mdiRecord;

  window = 0;
  sectionShowing = false;

  projects: Project[] = [
      {
          titleColor: "secondary--text",
          img: "fortheberry.png",
          arrowsColor: "white",

          text: ["projects.fortheberry.story", "projects.fortheberry.context"],

          codeImg: "itchio.png",
          codeLink: "https://rogue888.itch.io/for-the-berry",

          captions: ["projects.fortheberry.ide", "projects.fortheberry.mvp"],
      },
      {
          titleColor: "secondary--text",
          img: "shareyourfit.png",
          arrowsColor: "white",

          text: ["projects.shareyourfit.story", "projects.shareyourfit.context"],

          codeIcon: "devicon-gitlab-plain",
          codeLink: "https://gitlab.com/kilian-paquier/share-your-fit",
      },
      {
          titleColor: "secondary--text",
          img: "gestionepi.png",
          arrowsColor: "white",

          text: ["projects.gestionepi.story", "projects.gestionepi.context"],

          codeIcon: "devicon-gitlab-plain",
          codeLink: "https://gitlab.com/kilian-paquier/gestion-epi",

          captions: ["projects.gestionepi.epi", "projects.gestionepi.mvp"],
      },
      {
          title: "T.A.GEDOFIS",
          titleColor: "secondary--text",
          img: "tagedofis.png",
          arrowsColor: "secondary",

          text: ["projects.tagedofis.story", "projects.tagedofis.context"],

          codeIcon: "devicon-gitlab-plain",
          codeLink: "https://gitlab.com/kilian-paquier/tagedofis",

          captions: ["projects.tagedofis.ddt"],
      },
  ];

  redirectTo(url: string): void {
      window.open(url);
  }
}
</script>

<style lang="sass" scoped>
.skill-icon
    font-size: 2.25rem

a
    text-decoration: none

.cursor-pointer
    cursor: pointer
</style>
