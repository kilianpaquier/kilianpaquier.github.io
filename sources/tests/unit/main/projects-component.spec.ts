import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import ProjectsComponent from "@/components/main/ProjectsComponent.vue";
import Vuetify from "vuetify";
import Vue from "vue";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("ProjectsComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let wrapper: Wrapper<any>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(ProjectsComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create ProjectsComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
