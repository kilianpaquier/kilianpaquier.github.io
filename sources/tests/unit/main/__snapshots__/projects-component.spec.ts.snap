// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsComponent.vue should create ProjectsComponent 1`] = `
<div class="container container--fluid">
  <h2 class="font-weight-bold mb-3">$vuetify.projects-title</h2>
  <div class="v-window elevation-1 v-item-group theme--light v-window--show-arrows-on-hover" horizontal="">
    <div class="v-window__container">
      <transition-stub name="">
        <div class="v-window-item v-window-item--active">
          <div class="v-card v-sheet theme--light">
            <div class="v-image v-responsive align-end cursor-pointer theme--light">
              <div class="v-responsive__content">
                <!---->
                <div class="row pb-1">
                  <div class="d-flex col col-12">
                    <div class="mx-auto d-none d-sm-block">
                      <div class="v-item-group theme--light"><button type="button" class="v-btn v-btn--active v-btn--icon v-btn--round theme--light v-size--default white--text v-item--active"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button></div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <transition-stub name="expand-transition">
              <div style="display: none;">
                <div class="v-card__text">
                  <div class="row">
                    <div class="py-2 col col-12">$vuetify.projects.fortheberry.story</div>
                    <div class="py-2 col col-12">$vuetify.projects.fortheberry.context</div>
                  </div>
                  <div class="row">
                    <div class="text-center py-2 col col-12 mb-4">
                      <div class="v-image v-responsive mx-auto cursor-pointer theme--light" style="width: 50px;">
                        <div class="v-responsive__content"></div>
                      </div>
                      <!---->
                    </div>
                  </div>
                  <div class="row text-right no-gutters">
                    <div class="caption col col-12">$vuetify.projects.fortheberry.ide</div>
                    <div class="caption col col-12">$vuetify.projects.fortheberry.mvp</div>
                  </div>
                </div>
              </div>
            </transition-stub>
          </div>
        </div>
      </transition-stub>
      <transition-stub name="">
        <div class="v-window-item" style="display: none;">
          <div class="v-card v-sheet theme--light">
            <div class="v-image v-responsive align-end cursor-pointer theme--light">
              <div class="v-responsive__content">
                <!---->
                <div class="row pb-1">
                  <div class="d-flex col col-12">
                    <div class="mx-auto d-none d-sm-block">
                      <div class="v-item-group theme--light"><button type="button" class="v-btn v-btn--active v-btn--icon v-btn--round theme--light v-size--default white--text v-item--active"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button></div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <transition-stub name="expand-transition">
              <div style="display: none;">
                <div class="v-card__text">
                  <div class="row">
                    <div class="py-2 col col-12">$vuetify.projects.shareyourfit.story</div>
                    <div class="py-2 col col-12">$vuetify.projects.shareyourfit.context</div>
                  </div>
                  <div class="row">
                    <div class="text-center py-2 col col-12">
                      <!----> <a><em class="colored skill-icon devicon-gitlab-plain"></em></a>
                    </div>
                  </div>
                  <!---->
                </div>
              </div>
            </transition-stub>
          </div>
        </div>
      </transition-stub>
      <transition-stub name="">
        <div class="v-window-item" style="display: none;">
          <div class="v-card v-sheet theme--light">
            <div class="v-image v-responsive align-end cursor-pointer theme--light">
              <div class="v-responsive__content">
                <!---->
                <div class="row pb-1">
                  <div class="d-flex col col-12">
                    <div class="mx-auto d-none d-sm-block">
                      <div class="v-item-group theme--light"><button type="button" class="v-btn v-btn--active v-btn--icon v-btn--round theme--light v-size--default white--text v-item--active"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button></div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <transition-stub name="expand-transition">
              <div style="display: none;">
                <div class="v-card__text">
                  <div class="row">
                    <div class="py-2 col col-12">$vuetify.projects.gestionepi.story</div>
                    <div class="py-2 col col-12">$vuetify.projects.gestionepi.context</div>
                  </div>
                  <div class="row">
                    <div class="text-center py-2 col col-12 mb-4">
                      <!----> <a><em class="colored skill-icon devicon-gitlab-plain"></em></a>
                    </div>
                  </div>
                  <div class="row text-right no-gutters">
                    <div class="caption col col-12">$vuetify.projects.gestionepi.epi</div>
                    <div class="caption col col-12">$vuetify.projects.gestionepi.mvp</div>
                  </div>
                </div>
              </div>
            </transition-stub>
          </div>
        </div>
      </transition-stub>
      <transition-stub name="">
        <div class="v-window-item" style="display: none;">
          <div class="v-card v-sheet theme--light">
            <div class="v-image v-responsive align-end cursor-pointer theme--light">
              <div class="v-responsive__content">
                <div class="v-card__title secondary--text">T.A.GEDOFIS</div>
                <div class="row pb-1">
                  <div class="d-flex col col-12">
                    <div class="mx-auto d-none d-sm-block">
                      <div class="v-item-group theme--light"><button type="button" class="v-btn v-btn--active v-btn--icon v-btn--round theme--light v-size--default secondary--text v-item--active"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default secondary--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default secondary--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default secondary--text"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M19,12C19,15.86 15.86,19 12,19C8.14,19 5,15.86 5,12C5,8.14 8.14,5 12,5C15.86,5 19,8.14 19,12Z"></path></svg></span></span></button></div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <transition-stub name="expand-transition">
              <div style="display: none;">
                <div class="v-card__text">
                  <div class="row">
                    <div class="py-2 col col-12">$vuetify.projects.tagedofis.story</div>
                    <div class="py-2 col col-12">$vuetify.projects.tagedofis.context</div>
                  </div>
                  <div class="row">
                    <div class="text-center py-2 col col-12 mb-4">
                      <!----> <a><em class="colored skill-icon devicon-gitlab-plain"></em></a>
                    </div>
                  </div>
                  <div class="row text-right no-gutters">
                    <div class="caption col col-12">$vuetify.projects.tagedofis.ddt</div>
                  </div>
                </div>
              </div>
            </transition-stub>
          </div>
        </div>
      </transition-stub>
      <div class="v-window__prev"><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text" aria-label="Previous visual"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"></path></svg></span></span></button></div>
      <div class="v-window__next"><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default white--text" aria-label="Next visual"><span class="v-btn__content"><span aria-hidden="true" class="v-icon notranslate theme--light"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" class="v-icon__svg"><path d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"></path></svg></span></span></button></div>
    </div>
  </div>
</div>
`;
